src/material/snack-bar/snack-bar-container.ts

```diff
@@ -220,6 +231,54 @@ export abstract class _MatSnackBarContainerBase extends BasePortalOutlet impleme
         element.classList.add(panelClasses);
       }
     }
+
+    this._exposeToModals();
+  }
+
+  /**
+   * Some browsers won't expose the accessibility node of the live element if there is an
+   * `aria-modal` and the live element is outside of it. This method works around the issue by
+   * pointing the `aria-owns` of all modals to the live element.
+   */
+  private _exposeToModals() {
+    // TODO(crisbeto): consider de-duplicating this with the `LiveAnnouncer`.
+    // Note that the selector here is limited to CDK overlays at the moment in order to reduce the
+    // section of the DOM we need to look through. This should cover all the cases we support, but
+    // the selector can be expanded if it turns out to be too narrow.
+    const id = this._liveElementId;
+    const modals = this._document.querySelectorAll(
+      'body > .cdk-overlay-container [aria-modal="true"]',
+    );
+
+    for (let i = 0; i < modals.length; i++) {
+      const modal = modals[i];
+      const ariaOwns = modal.getAttribute('aria-owns');
+      this._trackedModals.add(modal);
+
+      if (!ariaOwns) {
+        modal.setAttribute('aria-owns', id);
+      } else if (ariaOwns.indexOf(id) === -1) {
+        modal.setAttribute('aria-owns', ariaOwns + ' ' + id);
+      }
+    }
+  }
+
+  /** Clears the references to the live element from any modals it was added to. */
+  private _clearFromModals() {
+    this._trackedModals.forEach(modal => {
+      const ariaOwns = modal.getAttribute('aria-owns');
+
+      if (ariaOwns) {
+        const newValue = ariaOwns.replace(this._liveElementId, '').trim();
```

Comment:
What happens if it would be in the middle of other ids? e.g. `aria-owns="existing <snackbar> some-other-added"`? I guess that is very unlikely to ever happen but want to see what you think.

Reply:
I think that would be fine since it would just leave a whitespace which the browser is going to ignore.